src/optsummary.jl

```diff
@@ -110,3 +110,6 @@ function NLopt.Opt(optsum::OptSummary)
     end
     opt
 end
+
+StructTypes.StructType(::Type{<:OptSummary}) = StructTypes.Mutable()
```

Comment:
Since `OptSummary` is a concrete type, there can't be subtypes and the subtype operator can't really do anything.

Reply:
The subtype operator is necessary because `OptSummary` is a parametric type.  The concrete types are `OptSummary{Float64}`, etc.  I had drafted an issue on whether it makes sense to continue to have `OptSummary` be a parametric type - it is closely tied to the `NLopt` package for which the only floating point type that can be used is `Float64` - but I didn't post it.

Anyway, the choices are to use the subtype operator here or to remove the parameter for `OptSummary` and I chose to keep the parameter.  It's a house of cards in a way - if we remove the parameter on `OptSummary` we might as well remove it everywhere.  I think it is better to retain the redundant parameter on `OptSummary` and modify that struct if we decide to incorporate other optimizers.

Reply:
Ah, yeah I had forgotten that `OptSummary` is parametric. I think it's worth keeping around the type parameter. On my long list of dream projects is changing the internals to take advantage of `MathOptInterface`, which I think NLopt finally supports. Then we could trivially support additional optimizers.